java/src/main/java/ai/onnxruntime/OrtSession.java

```diff
@@ -0,0 +1,494 @@
+                    throw new IllegalArgumentException("Unknown output name " + s + ", expected one of " + outputNames.toString());
+                }
+            }
+            OnnxValue[] outputValues = run(OnnxRuntime.ortApiHandle,nativeHandle, allocator.handle, inputNamesArray, inputHandles, numInputs, outputNamesArray, numOutputs);
+            return new Result(outputNamesArray,outputValues);
+        } else {
+            throw new IllegalStateException("Trying to score a closed OrtSession.");
+        }
+    }
+
+    @Override
+    public String toString() {
+        return "OrtSession(numInputs="+numInputs+",numOutputs="+numOutputs+")";
+    }
+
+    /**
+     * Closes the session, releasing it's resources.
+     * @throws OrtException If it failed to close.
+     */
+    @Override
+    public void close() throws OrtException {
+        if (!closed) {
+            closeSession(OnnxRuntime.ortApiHandle,nativeHandle);
+            closed = true;
+        } else {
+            throw new IllegalStateException("Trying to close an already closed OrtSession.");
+        }
+    }
+
+    private static Map<String,NodeInfo> wrapInMap(NodeInfo[] infos) {
+        Map<String,NodeInfo> output = new LinkedHashMap<>();
+
+        for (int i = 0; i < infos.length; i++) {
+            output.put(infos[i].getName(),infos[i]);
+        }
+
+        return output;
+    }
+
+    private native long createSession(long apiHandle, long envHandle, String modelPath, long optsHandle) throws OrtException;
+    private native long createSession(long apiHandle, long envHandle, byte[] modelArray, long optsHandle) throws OrtException;
+
+    private native long getNumInputs(long apiHandle, long nativeHandle) throws OrtException;
+    private native String[] getInputNames(long apiHandle, long nativeHandle, long allocatorHandle) throws OrtException;
+    private native NodeInfo[] getInputInfo(long apiHandle, long nativeHandle, long allocatorHandle) throws OrtException;
+
+    private native long getNumOutputs(long apiHandle, long nativeHandle) throws OrtException;
+    private native String[] getOutputNames(long apiHandle, long nativeHandle, long allocatorHandle) throws OrtException;
+    private native NodeInfo[] getOutputInfo(long apiHandle, long nativeHandle, long allocatorHandle) throws OrtException;
+
+    private native OnnxValue[] run(long apiHandle, long nativeHandle, long allocatorHandle, String[] inputNamesArray, long[] inputs, long numInputs, String[] outputNamesArray, long numOutputs) throws OrtException;
+
+    private native void closeSession(long apiHandle, long nativeHandle) throws OrtException;
+
+    /**
+     * Represents the options used to construct this session.
+     * <p>
+     * Used to set the number of threads, optimisation level, accelerator backend and other options.
+     * <p>
+     * Modifying this after the session has been constructed will have no effect.
+     */
+    public static class SessionOptions implements AutoCloseable {
```

Comment:
Should this be a standalone class?

Reply:
It's only useful in the context of creating the OrtSession, and it's essentially a Builder (and could become one at some point in the future), so keeping it as a public static class inside OrtSession shows where it's expected uses are.